# Copyright (c) 2006, 2009, Oracle and/or its affiliates. All rights reserved.
#
# Permission is hereby granted, free of charge, to any person obtaining a
# copy of this software and associated documentation files (the "Software"),
# to deal in the Software without restriction, including without limitation
# the rights to use, copy, modify, merge, publish, distribute, sublicense,
# and/or sell copies of the Software, and to permit persons to whom the
# Software is furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice (including the next
# paragraph) shall be included in all copies or substantial portions of the
# Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
# DEALINGS IN THE SOFTWARE.
#

Change needed to resume from S3 suspend on Rage XL chipset in Ultra 20.
https://bugs.freedesktop.org/show_bug.cgi?id=6829

diff --git a/src/aticonsole.c b/src/aticonsole.c
index 50cc6e1..947a122 100644
--- a/src/aticonsole.c
+++ b/src/aticonsole.c
@@ -515,6 +515,16 @@ ATIEnterGraphics
     ATIPtr      pATI
 )
 {
+    CARD32 PciReg;
+    pciVideoPtr pVideo = pATI->PCIInfo;
+
+/*
+* Possibly fix block I/O indicator in PCI configuration space.
+*/
+    PCI_READ_LONG(pVideo, &PciReg, PCI_REG_USERCONFIG);
+    if (!(PciReg & 0x00000004U))
+	PCI_WRITE_LONG(pVideo, (PciReg | 0x00000004U), PCI_REG_USERCONFIG);
+
     /* Map apertures */
     if (!ATIMapApertures(pScreenInfo->scrnIndex, pATI))
         return FALSE;
